pkgs/tools/text/ugrep/default.nix

```
@@ -0,0 +1,23 @@
+{ lib, fetchFromGitHub, stdenv, pcre2, boost, zlib, lzham }:
+
+stdenv.mkDerivation rec {
```

According to [this](https://nix.dev/anti-patterns/language.html#rec-expression), the `rec` expression is an anti-pattern.  It's probably easier to do
```suggestion
let
  pname = "ugrep";
  version = "2.4.1";
in stdenv.mkDerivation {
```